DaS-PC-MPChan/MainWindow.vb

```diff
@@ -257,6 +281,32 @@ Public Class MainWindow
             dgvFavoriteNodes.Rows.Add(key.GetValue(id), id)
         Next
     End Sub
+
+    'Read in blocked node list from register
+    Private Sub loadBlockedNodes()
+        Dim key As Microsoft.Win32.RegistryKey
+        key = My.Computer.Registry.CurrentUser.OpenSubKey("Software\DSCM\BlockedNodes", True)
+
+        If key IsNot Nothing Then
+            For Each id As String In key.GetValueNames()
+                dgvBlockedNodes.Rows.Add(id, key.GetValue(id))
+            Next
+        End If
+    End Sub
+
+    'Disconnect from blocked users every tick
+    Private Sub autoDisconnectBlockList_Tick() Handles autoDisconnectBlockList_Timer.Tick
+        autoDisconnectBlocked()
+    End Sub
+    Private Sub autoDisconnectBlocked()
+        If dsProcess IsNot Nothing Then
+            For Each entry As DataGridViewRow In dgvBlockedNodes.Rows
+                Dim steamid As String = entry.Cells(1).Value
+                dsProcess.DisconnectSteamId(steamid)
```

Comment:
This line will start a thread in the Dark Souls Process and send a packet to the user with the given steamID. It does this independent of whether we are actually connected to that user. This seems bad.

I am also not sure whether there are race conditions with this – starting 5 of these threads at once might cause problems.

Disconnects should only be triggered for nodes that we are actually connected to and then only one at a time.

Reply:
How do you feel about
```
        If dsProcess IsNot Nothing Then
            For Each blockNode As DataGridViewRow In dgvBlockedNodes.Rows
                For Each connectedNode As DataGridViewRow In dgvMPNodes.Rows
                    Dim blockSteamId As String = blockNode.Cells(1).Value
                    Dim connectedSteamID As String = connectedNode.Cells(1).Value
                    If (blockSteamId.Equals(connectedSteamID)) Then
                        dsProcess.DisconnectSteamId(blockSteamId)
                    End If
                Next
            Next
        End If
```
This fixes the issue with disconnecting non connected users. Although i'm not sure what you mean by disconnects should "only one at a time".